File: 백준/Silver/1543. 문서 검색/문서 검색.py
import sys
input = sys.stdin.readline

text = input().strip()
find_word = input().strip()

cnt = 0
idx = 0
while idx<=len(text)-len(find_word):
    if text[idx:idx+len(find_word)] == find_word:
        cnt +=1
        idx += len(find_word)
    else:
        idx +=1
print(cnt)

File: 백준/Silver/1543. 문서 검색/문서 검색.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String word = sc.nextLine();
        String find_word = sc.nextLine();

        int cnt = 0;
        int idx = 0;
        while(idx<=word.length()-find_word.length()){
            if(word.substring(idx, idx+find_word.length()).equals(find_word)){
                cnt +=1;
                idx += find_word.length();
            }else{
                idx +=1;
            }
        }
        System.out.println(cnt);
    }
}